_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -103,3 +103,117 @@ test.describe('Snapshot Repositories', () => {
+        await row.getByRole('menuitem', { name: 'Edit' }).click({ timeout: 60000 });
+        await page
+          .getByLabel('URL')
+          .fill(`https://fedorapeople.org/groups/katello/fakerepos/zoo${i}/`);
+        await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+        await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      });
+    }
+  });
+
+  await test.step('Verify the snapshot count for the repo.', async () => {
+    const row = await getRowByNameOrUrl(page, repoName);
+    await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    await page.getByRole('button', { name: 'Kebab toggle' }).click();
+    await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+    // Count the number of rows in the snapshot list table
+    const snapshotCount = (await page.getByTestId('snapshot_list_table').locator('tr').count()) - 1; // Subtract 1 for the header row
+    // Create a template which uses the repo and assert that is uses the latest snapshot
+    await page.getByLabel('Close', { exact: true }).click();
+    await navigateToTemplates(page);
+    await page.getByRole('button', { name: 'Add content template' }).click();
+    await page.getByRole('button', { name: 'Select architecture' }).click();
+    await page.getByRole('option', { name: 'aarch64' }).click();
+    await page.getByRole('button', { name: 'Select version' }).click();
+    await page.getByRole('option', { name: 'el9' }).click();
+    await page.getByRole('button', { name: 'Next', exact: true }).click();
+    await page.getByRole('button', { name: 'Next', exact: true }).click();
+    await row.getByRole('gridcell', { name: 'Select row' }).locator('label').click();
+    await page.getByRole('button', { name: 'Next', exact: true }).click();
+    await page.getByText('Use latest contentAlways use').click();
+    await page.getByRole('radio', { name: 'Use latest content' }).check();
+    await page.getByRole('button', { name: 'Next' }).click();
+    await page.getByPlaceholder('Enter name').fill(`${templateName}`);
+    await page.getByPlaceholder('Description').fill('Template test');
+    await page.getByRole('button', { name: 'Next', exact: true }).click();
+    await page.getByRole('button', { name: 'Create other options' }).click();
+    await page.getByText('Create template only', { exact: true }).click();
+    const templateRow = await getRowByNameOrUrl(page, templateName);
+    await expect(templateRow.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    // Verify the template is created and uses the latest snapshot
+    await expect(
+      await page.getByRole('gridcell', { name: 'Use latest' }).first().textContent(),
+    ).toBe('Use latest');
+    // Assert that the template snapshot count matches the repo snapshot count
+    expect(snapshotCount).toBe(4);
+
+    // Test deletion of a single snapshot.
+    await test.step('Delete a single snapshot', async () => {
```

Comment:
this step (and the bulk delete one) is nested inside of another one (`Verify snapshot count for the repo`), it doesn't really make sense for it to be there from the test point of view and in general I would not nest any steps 😅

Reply:
I understood what are you suggesting made the changes and seperated it in different test step now 
thanks !